pkg/querier/queryrange/codec.go

```diff
@@ -1962,10 +1965,27 @@ func mergeLokiResponse(responses ...queryrangebase.Response) *LokiResponse {
 		lokiResponses = make([]*LokiResponse, 0, len(responses))
 	)
 
+	uniqueWarnings := map[string]struct{}{}
 	for _, res := range responses {
 		lokiResult := res.(*LokiResponse)
 		mergedStats.MergeSplit(lokiResult.Statistics)
 		lokiResponses = append(lokiResponses, lokiResult)
+
```

Comment:
Loki splits a query and then merges the responses into one. Here we make sure all the warnings from subqueries make it into the merged query. We'll do the same below for metrics queries